server/handler/fetcher.go

```diff
@@ -152,9 +153,26 @@ func (cf *ConfigFetcher) fetchConfigContents(ctx context.Context, client *github
 
 	file, _, _, err := client.Repositories.GetContents(ctx, owner, repo, path, opts)
 	if err != nil {
-		if rerr, ok := err.(*github.ErrorResponse); ok && rerr.Response.StatusCode == http.StatusNotFound {
+		rerr, ok := err.(*github.ErrorResponse)
+		if ok && rerr.Response.StatusCode == http.StatusNotFound {
 			return nil, nil
 		}
+		if ok && cf.hasTooLargeError(rerr) {
+			// GetContents only supports file sizes up to 1 MB, DownloadContents supports files up to 100 MB (with an additional API call)
+			reader, err := client.Repositories.DownloadContents(ctx, owner, repo, path, opts)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to download content of %s/%s@%s/%s", owner, repo, ref, path)
+			}
+			downloadedContent, readErr := ioutil.ReadAll(reader)
+			closeError := reader.Close()
```

Comment:
I think it's a bit more idiomatic to handle this in a `defer` and either set a named return value or log it as a warning/error:

```go
reader, err := client.Repositories.DownloadContents(ctx, owner, repo, path, opts)
if err != nil {
  return nil, errors.Wrapf(err, "failed to download content of %s/%s@%s/%s", owner, repo, ref, path)
}

defer func() {
  if cerr := reader.Close(); cerr != nil {
    logger.Error().Err(cerr).Msgf(...)
  }
}()

content, err := ioutil.ReadAll(reader)
// ...
```

I'd go for the log over returning an error since I don't think a failure to close this reader should block further evaluation of the request.

Reply:
Makes sense, thanks for the advice. I agree!